### ISSUE TITLE
[3.0]: Cannot Like

### ISSUE THREAD

**Issue by DiegoAndresCortes** [SimpleMachines/SMF#8115]
### Basic Information

Fatal error: Uncaught TypeError: Cannot assign string to property SMF\Actions\Like::$id_topic of type int in \Sources\Actions\Like.php:357

https://github.com/SimpleMachines/SMF/blob/ead27e998d9a3e9afa6e04f4cfbcfa473de00cc7/Sources/Actions/Like.php#L357

**Comment by Oldiesmann** [comment]
That sounds like the query generated an error rather than returning a result.

**Comment by DiegoAndresCortes** [comment]
In SMF 2.1 it returns the same output:
`array(2) { [0]=> string(1) "7" [1]=> string(1) "2" } `

SMF 3.0:
` array(2) { [0]=> string(1) "9" [1]=> string(1) "2" } `

Obviously they're different msg_ids.
The difference is that in 2.1 $this->_idTopic didn't have a type of course.

**Comment by vicram10** [comment]
Making a modification so that it first captures the value and then converts it to int would resolve the issue.

```
list($topic_id, $topicOwner) = Db::$db->fetch_row($request);
$this->id_topic = intval($topic_id);
```

By the way, it reproduces the same error

**Comment by Oldiesmann** [comment]
Completely forgot about this issue until now but this was fixed in #8230
